test/parser/parse.test.ts

```diff
@@ -910,6 +910,26 @@ See more help with --help`)
           delete process.env.TEST_FOO
         })
       }
+
+      it('ignores unset environment variables', async () => {
+        delete process.env.TEST_FOO
+        const out = await parse([], {
+          flags: {
+            foo: flags.boolean({env: 'TEST_FOO'}),
+          },
+        })
+        expect(out.flags.foo).to.be.undefined
+      })
+
+      it('uses default when environment variable is unset', async () => {
+        delete process.env.TEST_FOO
+        const out = await parse([], {
+          flags: {
+            foo: flags.boolean({env: 'TEST_FOO', default: true}),
+          },
+        })
+        expect(out.flags.foo).to.be.true
+      })
```

Comment:
Just to be 100% clear, both of these tests fail when run against `main`, because in both cases the absent env var is interpreted as falsey:

```
  2 failing

  1) parse
       env
         boolean
           ignores unset environment variables:
     AssertionError: expected false to be undefined
      at Context.<anonymous> (test/parser/parse.test.ts:921:36)

  2) parse
       env
         boolean
           uses default when environment variable is unset:

      AssertionError: expected false to be true
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (test/parser/parse.test.ts:931:36)
```